src/dotnet/Chat.Service/ChatsBackend.cs

```diff
@@ -823,6 +852,10 @@ await dbContext.Roles
 
         Chat ApplyDiff(Chat originalChat, ChatDiff? diff) {
             // Update
+            if (diff?.UserLinkId != null)
+                diff = diff with {
+                    UserLinkId = diff.UserLinkId.Value.ToLower()
```

Comment:
is it kind of normalization? then how about an ext method for it?

Reply:
ToLower method is already a normalization method but with more specific name.
What do you suggest to do?